src/toast/ToastProvider.tsx

```diff
@@ -26,13 +27,25 @@ interface ToastContextProviderProps {
 function ToastContextProvider({
   children,
   customRootId,
-  autoCloseToasts = true
+  autoCloseToasts = true,
+  limit
 }: ToastContextProviderProps) {
   const [state, dispatch] = useReducer(toastReducer, {
     ...initialToastState,
-    autoCloseToasts
+    autoCloseToasts,
+    limit
   });
 
+  useEffect(() => {
+    if (limit) {
```

Comment:
Kind of an extreme edge case but what if they provide limit as 0 ?

Reply:
Updated this statement with `limit !== undefined`